### PR TITLE
initial sample

### PR DESCRIPTION
**Pull request by derisen**
### Overview

This sample demonstrates a React & Redux single-page application allowing a user to authenticate and authorize an ASP.NET Core Web API that was protected by [Azure AD](https://azure.microsoft.com/en-ca/services/active-directory/) to call [MS Graph API](https://developer.microsoft.com/en-us/graph) on its behalf using the AAD [on-behalf-of flow](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-on-behalf-of-flow).

The API call to MS Graph API is made using the [MS Graph SDK](https://docs.microsoft.com/en-us/graph/sdks/sdks-overview).

### Scenario

- The sample implements an **onboarding** scenario where a profile is created for a new user upon first login. The profile fields are pre-populated by the available information about the user on MS Graph API.
- ProfileSPA uses [MSAL.js](https://github.com/AzureAD/microsoft-authentication-library-for-js) to authenticate a user and [React-Redux](https://react-redux.js.org/) to store id and access tokens.
- Once the user authenticates, ProfileSPA obtains an [access token](https://docs.microsoft.com/azure/active-directory/develop/access-tokens) from Azure Active Directory (Azure AD).
- The access token is then used to authorize the ProfileAPI to call MS Graph API **on user's behalf**. In order to call MS Graph API, ProfileAPI uses the [MS Graph SDK](https://docs.microsoft.com/en-us/graph/sdks/sdks-overview).
- To protect its endpoint and accept only the authorized calls, the ProfileAPI uses [MSAL.NET](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet) and [Microsoft.Identity.Web](https://github.com/AzureAD/microsoft-identity-web).

Some screenshots:

login
consent
onboard
view

### CODE REVIEW

**Comment by franva** [comment]
This is what I am looking for, thanks a lot. I will give it a try.

File: README.md [hunk 1]
```diff
@@ -1,53 +1,280 @@
...
+
+##### Configure the client app (ProfileSPA) to use your app registration
+
+>In the steps below, "ClientID" is the same as "Application ID" or "AppId".
+
+1. Open the `ProfileSPA\src\utils\authConfig.js` file
+1. Find the app key `clientId` and replace the existing value with the application ID (clientId) of the `ProfileAPIandSPA` application copied from the Azure portal.
+1. Find the app key `redirectUri` and replace the existing value with the base address of the ProfileSPA project (by default `http://localhost:3000/`).
+1. Find the app key `resourceUri` and replace the existing value with the base address of the ProfileAPI project (by default `https://localhost:44351/api/profile/`).
+1. Find the app key `resourceScope` and replace the existing value with *Scope* you created earlier `https://{tenant-domain}/{app-name}`.
```
Comment (franva) [inline README.md]: Find the app key `resourceScope` and replace the existing value with *Scope* you created earlier `https://{tenant-domain}/{app-name}/access_as_user`.

Comment (derisen) [reply README.md]: good catch, thanks!

**Review by TiagoBrenck — APPROVED** [review]
Just an observation in the app scripts, on the SPA project, authConfig.js is receiving the app id of the API instead of the SPA. Other than this, LGTM

File: AppCreationScripts/sample.json [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      "ManualSteps": [
+        {
+          "Comment": "Navigate to the Manifest page and change 'signInAudience' to 'AzureADandPersonalMicrosoftAccount'."
+        },
+        {
+          "Comment": "Navigate to the Manifest page and change 'accessTokenAcceptedVersion' to 2."
+        },
+        {
+          "Comment": "Navigate to the Manifest page, find the entry for 'KnownClientApplications' and add the clientId of ProfileSPA."
+        }
```
Comment (kalyankrishna1) [inline AppCreationScripts/sample.json:33]: This is not needed. The generated code in configure.ps1 does this as well

**Review by kalyankrishna1 — APPROVED** [review]
:shipit:

**Pull request merged**